x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/secrets/action/TransportPostSecretAction.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector.secrets.action;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.client.internal.OriginSettingClient;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.util.concurrent.EsExecutors;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+
+import static org.elasticsearch.xcontent.XContentFactory.jsonBuilder;
+import static org.elasticsearch.xpack.application.connector.secrets.SecretsIndexService.CONNECTOR_SECRETS_INDEX_NAME;
+import static org.elasticsearch.xpack.core.ClientHelper.CONNECTORS_ORIGIN;
+
+public class TransportPostSecretAction extends HandledTransportAction<PostSecretRequest, PostSecretResponse> {
+
+    private final Client client;
+
+    @Inject
+    public TransportPostSecretAction(TransportService transportService, ActionFilters actionFilters, Client client) {
+        super(PostSecretAction.NAME, transportService, actionFilters, PostSecretRequest::new, EsExecutors.DIRECT_EXECUTOR_SERVICE);
+        this.client = new OriginSettingClient(client, CONNECTORS_ORIGIN);
+    }
+
+    protected void doExecute(Task task, PostSecretRequest request, ActionListener<PostSecretResponse> listener) {
+        try {
+            client.prepareIndex(CONNECTOR_SECRETS_INDEX_NAME)
```

Comment:
Sorry, I don't understand exactly what you mean. Can you expand on this a bit?

Reply:
Instead of doing the actual index operation here, I would defer to an operation created in the `SecretIndexService` for that. That is the class that should deal with all operations related to storage and retrieval of the secrets (like a DAO pattern) and understands the internals of the index.

Reply:
++ here are two examples from @jedrazb and me: [ConnectorIndexService](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java) and [ConnectorSyncJobIndexService](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexService.java), hope this helps 🙏

Reply:
Thanks for the clarifications! I've attempted to do this in this commit: https://github.com/elastic/elasticsearch/pull/103683/commits/34c9298711e1b205d353c1aa82d88dd1fc5b55b5

Is this what you're talking about?

Reply:
Yes, exactly, thanks for the change! Small nit: IMHO `createSecret` is a better name than `postSecret` as `post` comes from the REST/HTTP layer. I left a [similar comment](https://github.com/elastic/elasticsearch/pull/103977#discussion_r1446246921) on one of Jedr's PRs, where I describe the rationale in more detail. Sorry for the nitpicking, but I think naming consistency can help us all in the future navigating the module :)

Reply:
It should be create, we don't support updating secrets

Reply:
This solves it for me, thanks @navarone-feekery ! Agree on the naming nit, maybe `updateSecret` would be better. But, you know, _naming_...

Reply:
I changed it to `createSecret` in [this commit](https://github.com/elastic/elasticsearch/pull/103683/commits/9cf3f81a9e115dd26c15e24124d087a944ea4543).